polar-core/src/formatting.rs

```diff
@@ -688,6 +692,14 @@ mod tests {
             007:        seven"};
         assert_eq!(lines, expected, "\n{}", lines);
 
+        let lines = source_lines(&source, 1, 2);
+        let expected = indoc! {"
+            001:  one
+                  ^
+            002:   two
+            003:    three"};
```

Comment:
previously this would print
```
001:  one
      ^
002:   two
003:    three
004:     four
005:      five
```
b/c we were unconditionally asking for up to `context_lines * 2` lines of context. If `target_line` were line 0 and we'd asked for `2` lines of context on either side of the target line, we'd print 5 lines instead of 3.